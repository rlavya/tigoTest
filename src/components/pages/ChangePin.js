import React, { Component } from "react";
import RaisedButton from "material-ui/RaisedButton";
import TextField from "material-ui/TextField";
import PageTitle from "../components/PageTitle";
import { lightBlue200, indigo300 } from "material-ui/styles/colors";
import get from "lodash/get";
import { connect } from "react-redux";
import {
  updateNewPinAction,
  updateOldPinAction,
  changePinAction
} from "../../actionCreators/auth.js";
import { onEnterAction } from "../../util";
import styled from "styled-components";

const InputWrap = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  img {
    display: block;
    margin-bottom: -23px;
    margin-right: 10px;
    margin-left: -30px;
  }
`;

export class ChangePin extends Component {
  render() {
    const changePinIfValid = onEnterAction(() => {
      if (this.props.oldPin.trim() !== "" && this.props.newPin.trim() !== "") {
        this.props.changePin();
      }
    });

    return (
      <PageTitle page="ChangePin">
        <div className="max-width-4 mx-auto">
          <div className="clearfix center">
            <h1 className="light text-primary h2">Ingresa tus credenciales:</h1>

            <div className="masked-input">
              <InputWrap>
                <img src="/images/icn-password.png" alt="" />
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN ACTUAL"
                  type="tel"
                  pattern="[0-9]*"
                  id="oldPin"
                  value={this.props.oldPin}
                  onChange={this.props.updateOldPin}
                  onKeyDown={changePinIfValid}
                />
              </InputWrap>
            </div>
            <br />
            <div className="masked-input">
              <InputWrap>
                <img src="/images/icn-password-new.png" alt="" />
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN NUEVO"
                  type="tel"
                  pattern="[0-9]*"
                  id="newPin"
                  value={this.props.newPin}
                  onChange={this.props.updateNewPin}
                  onKeyDown={changePinIfValid}
                />
              </InputWrap>
            </div>
            <br />

            <RaisedButton
              className="mt2"
              secondary={true}
              label="Cambiar"
              onClick={this.props.changePin}
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={
                this.props.oldPin.trim() === "" ||
                this.props.newPin.trim() === ""
              }
            />
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      oldPin: get(state, "auth.pinChange.old") || "",
      newPin: get(state, "auth.pinChange.new") || ""
    };
  },
  dispatch => {
    return {
      updateOldPin: (e, value) => {
        return dispatch(updateOldPinAction(value));
      },
      updateNewPin: (e, value) => {
        return dispatch(updateNewPinAction(value));
      },
      changePin: (e, value) => {
        return dispatch(changePinAction());
      }
    };
  }
)(ChangePin);
