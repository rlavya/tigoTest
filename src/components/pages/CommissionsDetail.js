import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import { redirectTo } from "../../router";
import TransactionsOrCommissionsDetail from "../components/TransactionsOrCommissionsDetail";

export default connect(
  state => {
    const items = dataForId(
      state,
      configToId("CommissionDetail", "list", true)
    );
    return {
      items,
      pageName: siteTree.TransactionsDetail.name,
      periodFilter: state.route.periodFilter || "hora",
      transactionFilter: state.route.transactionFilter || "todas"
    };
  },
  dispatch => {
    return {
      changePeriodFilter(filter) {
        redirectTo({ periodFilter: filter }, true);
      },
      changeTransactionFilter(filter) {
        redirectTo({ transactionFilter: filter }, true);
      }
    };
  }
)(TransactionsOrCommissionsDetail);
