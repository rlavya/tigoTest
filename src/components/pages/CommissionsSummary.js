import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import TransactionsOrCommissionsSummary from "../components/TransactionsOrCommissionsSummary";

export default connect(state => {
  const commissions = dataForId(state, configToId("Commission", "list", true));

  return {
    items: commissions,
    pageName: siteTree.CommissionsSummary.name,
    detailViewPageName: siteTree.CommissionsDetail.name
  };
})(TransactionsOrCommissionsSummary);
