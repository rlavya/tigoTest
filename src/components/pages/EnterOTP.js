import React, { Component } from "react";
import PageTitle from "../components/PageTitle";
import { lightBlue200 } from "material-ui/styles/colors";
import RaisedButton from "material-ui/RaisedButton";
import get from "lodash/get";
import { connect } from "react-redux";
import TextField from "material-ui/TextField";
import styled from "styled-components";

import {
  requestOtpAction,
  updateOTPCodeAction,
  validateOTPCodeAction
} from "../../actionCreators/auth.js";

import { onEnterAction } from "../../util";

const TextFieldWrap = styled.div`
  margin: 0 auto;
  width: 157px;
`;

export class EnterOTP extends Component {
  render() {
    const onEnter = onEnterAction(() => {
      this.props.validateCode();
    });
    return (
      <PageTitle page="OTP">
        <div className="max-width-4 mx-auto">
          <div className="center">
            <h1 className="light text-primary h2">
              Ingresa el código <br />de verificación
            </h1>
            <div className="mt3">
              <p className="regular text-primary inline">
                Te enviamos un SMS al{" "}
              </p>
              <p className="bold text-primary inline">+{this.props.msisdn}</p>
            </div>
            <TextFieldWrap>
              <TextField
                value={this.props.otpCode}
                onChange={this.props.updateCode}
                id="validate-otp"
                className="mt4 mb1"
                type="tel"
                pattern="[0-9]*"
                fullWidth={true}
                onKeyDown={onEnter}
                underlineStyle={{
                  backgroundImage: "url(/images/segmented_inactive.png)",
                  backgroundRepeat: "no-repeat",
                  backgroundPosition: "2px 0px",
                  height: "2px",
                  border: "none",
                  bottom: "1px"
                }}
                underlineFocusStyle={{
                  backgroundImage: "url(/images/segmented_active.png)",
                  backgroundRepeat: "no-repeat",
                  backgroundPosition: "2px 0px",
                  height: "2px",
                  border: "none",
                  bottom: "1px"
                }}
                inputStyle={{
                  fontSize: "2.5rem",
                  fontFamily: "'Roboto Mono', monospace",
                  letterSpacing: "2px"
                }}
              />
            </TextFieldWrap>
            <br />
            <RaisedButton
              label="Continuar"
              secondary={true}
              className="mt2 mb3"
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={this.props.otpCode.length < 6}
              onClick={this.props.validateCode}
            />
            <br />
            <p className="inline text-primary mt2">¿No recibiste el código?</p>
            <p
              className="inline pointer"
              style={{ color: lightBlue200 }}
              onClick={this.props.requestNewCode}
            >
              {" "}
              Enviar de nuevo.
            </p>
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      msisdn: get(state, "auth.otp.msisdn") || "",
      otpCode: get(state, "auth.otpCode") || ""
    };
  },
  dispatch => {
    return {
      updateCode: (e, value) => {
        return dispatch(updateOTPCodeAction((value || "").trim()));
      },
      validateCode: () => {
        return dispatch(validateOTPCodeAction());
      },
      requestNewCode: () => {
        return dispatch(requestOtpAction({}));
      }
    };
  }
)(EnterOTP);
