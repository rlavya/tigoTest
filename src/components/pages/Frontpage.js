import React, { Component } from "react";
import { List, ListItem } from "material-ui/List";
import ChevronRight from "material-ui/svg-icons/navigation/chevron-right";
import Divider from "material-ui/Divider";
import { redirectTo } from "../../router";

export class Frontpage extends Component {
  render() {
    return (
      <div className="max-width-4 mx-auto bg-primary height100 flex flex-column">
        <img
          src="http://via.placeholder.com/250x250"
          width="250"
          height="250"
          className="mx-auto my2"
          alt=""
        />
        <List>
          <Divider style={{ backgroundColor: "#406A9C" }} />
          <ListItem
            primaryText={
              <div>
                <b>PDV</b> - Punto de Venta
              </div>
            }
            rightIcon={<ChevronRight color="#4CC5F2" />}
            onClick={() => {
              redirectTo({ page: "Login" });
            }}
            style={{ color: "#fafafa" }}
          />
          <Divider style={{ backgroundColor: "#406A9C" }} />
          <ListItem
            primaryText={
              <div>
                <b>PDA</b> - Punto de Activaci√≥n
              </div>
            }
            rightIcon={<ChevronRight color="#4CC5F2" />}
            onClick={() => {
              redirectTo({ page: "Login" });
            }}
            style={{ color: "#fafafa" }}
          />
          <Divider style={{ backgroundColor: "#406A9C" }} />
        </List>
      </div>
    );
  }
}

export default Frontpage;
