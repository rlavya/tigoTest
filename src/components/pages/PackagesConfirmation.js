import React, { Component } from "react";
import FlatButton from "material-ui/FlatButton";
import Divider from "material-ui/Divider";
import { getPackageById, formatPhoneNumber } from "../../util/index";
import get from "lodash/get";
import { connect } from "react-redux";
import styled from "styled-components";
import { clearCurrentUserAction } from "../../actionCreators/selling";

const BluePanel = styled.div`
  background: #00377b;
  height: 100%;
  overflow-x: hidden;
  overflow-y: scroll;
  -webkit-overflow-scrolling: touch;
`;

export class PackagesConfirmation extends Component {
  render() {
    return (
      <BluePanel className="max-width-4 mx-auto">
        <div className="clearfix center">
          <img
            alt=""
            src="/images/icn-sucess-green.png"
            className="mt2"
            style={{
              width: "35px",
              height: "35px"
            }}
          />
          <h1 className="h5 text-white text-big regular mb3 mt1">
            ¡Compra realizada<br />con éxito!
          </h1>
          <Divider
            style={{
              background: "#406A9C"
            }}
          />

          <h2 className="h6 regular text-tertiary p0 mt3 mb0">Paquete</h2>
          <h3 className="regular h5 text-white p0 m0">
            {this.props.package.name}
          </h3>
          <h3 className="regular h5 text-tertiary  p0 m0">
            {this.props.package.description}
          </h3>

          <h2 className="h6 regular text-tertiary p0 mt2 mb0">Valor</h2>
          <h3 className="regular h5 text-white text-big p0 m0">
            {this.props.package.currency}
            {this.props.package.displayPrice}
          </h3>

          <h2 className="h6 regular text-tertiary p0 mt2 mb0">
            Teléfono del usuario
          </h2>
          <h3 className="regular h5 text-white text-big p0 m0 mb2">
            {this.props.clientPhoneNumber}
          </h3>

          <Divider
            style={{
              background: "#406A9C"
            }}
          />
          <br />
          <FlatButton
            secondary={true}
            label="Volver Al Inicio"
            onClick={this.props.clear}
          />
        </div>
      </BluePanel>
    );
  }
}

export default connect(
  state => {
    const pack = getPackageById(state, get(state, "route.id"));

    const clientPhoneNumber = formatPhoneNumber(
      get(state, "clientPhone.phoneNumber")
    );

    return {
      clientPhoneNumber,
      package: pack
    };
  },
  dispatch => {
    return {
      clear: () => dispatch(clearCurrentUserAction())
    };
  }
)(PackagesConfirmation);
