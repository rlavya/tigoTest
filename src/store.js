import { applyMiddleware, createStore } from "redux";
import { createLogger } from "redux-logger";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import thunk from "redux-thunk";
import reducer from "./reducer";

const getMiddleware = () => {
  return applyMiddleware(createLogger(), thunk);
};

const store = createStore(reducer, composeWithDevTools(getMiddleware()));

export default store;
