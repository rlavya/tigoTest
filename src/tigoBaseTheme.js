const tigoBaseTheme = {
  fontFamily: "Roboto, sans-serif",
  borderRadius: 2,
  palette: {
    primary1Color: "#00377B",
    primary2Color: "#4CC5F2",
    primary3Color: "#333",
    accent1Color: "#4CC5F2",
    accent2Color: "#4CC5F2",
    accent3Color: "#4CC5F2",
    textColor: "#00377B",
    secondaryTextColor: "#839ebf",
    alternateTextColor: "#fafafa",
    canvasColor: "#FFFFFF"
  }
};

export default tigoBaseTheme;
