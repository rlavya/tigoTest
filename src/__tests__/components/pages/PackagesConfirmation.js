import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import tigoBaseTheme from "../../../tigoBaseTheme";
import { PackagesConfirmation } from "../../../components/pages/PackagesConfirmation";
import renderer from "react-test-renderer";
import React from "react";
import store from "../../../store";
import { Provider } from "react-redux";

it("renders correctly", () => {
  const rendered = renderer.create(
    <MuiThemeProvider muiTheme={getMuiTheme(tigoBaseTheme)}>
      <Provider store={store}>
        <PackagesConfirmation
          items={{ state: "loaded" }}
          package={{name: 'TEST'}}
        />
      </Provider>

    </MuiThemeProvider>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
