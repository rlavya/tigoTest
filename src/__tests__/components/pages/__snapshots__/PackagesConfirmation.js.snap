// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-width-4 mx-auto sc-bdVaJa bNDmez"
>
  <div
    className="clearfix center"
  >
    <svg
      className="mt2"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "#00377B",
          "display": "inline-block",
          "fill": "#5BAB4B",
          "height": "35px",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": "35px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
    <h1
      className="h5 text-white text-big regular mb3"
    >
      ¡Compra realizada
      <br />
      con éxito!
    </h1>
    <hr
      style={
        Object {
          "background": "#406A9C",
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
    <h2
      className="h6 regular text-tertiary p0 mt3 mb0"
    >
      Paquete
    </h2>
    <h3
      className="regular h5 text-white p0 m0"
    >
      TEST
    </h3>
    <h3
      className="regular h5 text-tertiary  p0 m0"
    />
    <h2
      className="h6 regular text-tertiary p0 mt2 mb0"
    >
      Valor
    </h2>
    <h3
      className="regular h5 text-white text-big p0 m0"
    />
    <h2
      className="h6 regular text-tertiary p0 mt2 mb0"
    >
      Telefono del cliente
    </h2>
    <h3
      className="regular h5 text-white text-big p0 m0 mb2"
    />
    <hr
      style={
        Object {
          "background": "#406A9C",
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
    <br />
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "#4CC5F2",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": 88,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          Volver Al Inicio
        </span>
      </div>
    </button>
  </div>
</div>
`;
